test/sets.jl

```diff
@@ -558,6 +558,11 @@ end
     @test allunique(x for x in [0.0, -0.0] if true)
     @test !allunique([NaN, NaN])
     @test !allunique(x for x in [NaN, NaN] if true)
+    # strings
+    @test !allunique("aaaa")
```

Comment:
At present we have:
```julia
julia> allunique("óá")
true

julia> allunique("óá")
false

julia> collect("óá")
2-element Vector{Char}:
 'ó': Unicode U+00F3 (category Ll: Letter, lowercase)
 'á': Unicode U+00E1 (category Ll: Letter, lowercase)

julia> collect("óá")
4-element Vector{Char}:
 'o': ASCII/Unicode U+006F (category Ll: Letter, lowercase)
 '́': Unicode U+0301 (category Mn: Mark, nonspacing)
 'a': ASCII/Unicode U+0061 (category Ll: Letter, lowercase)
 '́': Unicode U+0301 (category Mn: Mark, nonspacing)
```
What answers do we want? These ones match `unique`. 

If they are right, they should probably be tested.

(I know this isn't what this PR wanted to do, but it does add a String method.)